esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2315,9 +2315,33 @@ bool Cws_accessEx::onAddViewColumn(IEspContext &context, IEspAddViewColumnReques
 
         checkUser(context);
 
+        const char* origFileName = req.getFilename();
+        const char* origColumnName = req.getColumnname();
+
+        int fileLength = strlen(origFileName);
+        int columnLength = strlen(origColumnName);
+
+        if (fileLength == 0 || columnLength ==0)
+        {
+            resp.setSuccess(false);
+            resp.setResult("Filename or Columnname cannot be empty");
+            return false;
+        }
+
+        // All view column filename MUST be a full path including the scope, with the leading tilde (~)
+        // If filename doesn't start with a leading tilde, add it
+        StringBuffer normalizedFileName;
+
+        if (origFileName[0] != '~')
+        {
+            normalizedFileName.append('~'); 
```

Comment:
Please make sure your testing includes all permutations of scoped and nonscoped  LFNs